pkg/config/setup/system_probe.go

```diff
@@ -286,6 +287,9 @@ func InitSystemProbeConfig(cfg pkgconfigmodel.Config) {
 	// connection aggregation with port rollups
 	cfg.BindEnvAndSetDefault(join(netNS, "enable_connection_rollup"), false)
 
+	// network_config network path config
+	cfg.BindEnvAndSetDefault(join(npNS, "enabled"), false)
```

Comment:
Is this going to control something in system-probe?

Reply:
@brycekahle It's right now used to configuration NpScheduler which runs inside the process-agent.

From existing code, I see cases where process-agent checks depend on system-probe.yaml.

Do you see a better place where we can put those configurations?

Reply:
@brycekahle In Network Path (traceroute) case, there are some dependency with NPM ConnectionsCheck, it uses the connections to decide what to traceroute. 

I was thinking that since NPM ConnectionsCheck uses configs from system-probe.yaml, it made sense to also rely on it for Network Path. For example here, some configs used by ConnectionsCheck depending on system-probe.yaml:
- https://github.com/DataDog/datadog-agent/blob/c3b3a5d5154e31db2d92f6b420d993d9bdbcc12b/pkg/process/checks/net.go#L143-L144
- https://github.com/DataDog/datadog-agent/blob/c3b3a5d5154e31db2d92f6b420d993d9bdbcc12b/pkg/process/checks/net.go#L117-L119

Also, Network Path also depend on the `traceroute` module to be enabled to work (`traceroute.enabled: true` in `system-probe.yaml`)

The architecture look like this: 
![image](https://github.com/DataDog/datadog-agent/assets/49917914/9ad0e3db-3c6c-4978-9ebd-0f9b5eb53eff)


But I understand the rational you detailed above, if the relations above is not enough to justify having Network Path configs in system-probe.yaml, I'm not against moving configs to datadog.yaml instead (there is no process-agent.yaml right?).

@hmahmood @ken-schneider @leeavital Any thoughts on this?

Reply:
> there is no process-agent.yaml right?

There is not. The only reason `system-probe.yaml` exists is because of security reasons. It controls a privileged process, so it has different permissions from `datadog.yaml`.

Reply:
@brycekahle I moved the configs to datadog.yaml here 2887ede563295570772260cac4630e144c953dfa and ac08ffc1f76a2b6f95242f524506ea06640bd019 (second stacked PR), could you have another look?